vpr/src/route/rr_graph_generation/rr_node_indices.h

```diff
@@ -0,0 +1,66 @@
+#pragma once
+
+#include "rr_graph_builder.h"
+#include "rr_graph_view.h"
+#include "device_grid.h"
+#include "rr_types.h"
+#include "rr_graph_type.h"
+#include "rr_graph_utils.h"
+#include "clustered_netlist_fwd.h"
+
+/**
+ * @brief Allocates and populates data structures for efficient rr_node index lookups.
+ *
+ * This function sets up the `rr_node_indices` structure, which maps a physical location
+ * and type to the index of the first corresponding rr_node.
```

Comment:
Good to comment the parameters if you can.  
index is a bit baffling ... not sure what it does.  
nodes_per_chan is also not well documented. The t_chan_width data structure has no comments, so commenting it would be good (if you know what it does) and commenting what nodes_per_chan is here would be good.